packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -8450,220 +8452,414 @@ CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
     }
     }
 
+    /*********************************************************************/
     /**** 3) Copy all of the Same/Permute/Remote data into CSR_arrays ****/
     /*********************************************************************/
 
-    // Backwards compatibility measure.  We'll use this again below.
-
-    // TODO JHU Need to track down why numImportPacketsPerLID_ has not been corrently marked as modified on host (which it has been)
-    // TODO JHU somewhere above, e.g., call to Distor.doPostsAndWaits().
-    // TODO JHU This only becomes apparent as we begin to convert TAFC to run on device.
-    destMat->numImportPacketsPerLID_.modify_host(); //FIXME
+    bool runOnHost = std::is_same_v<typename device_type::execution_space, Kokkos::Serial> && !useKokkosPath;
```

Comment:
```suggestion
    bool runOnHost = std::is_same_v<typename device_type::memory_space, Kokkos::HostSpace> && !useKokkosPath;
```
Kokkos::Serial doesn't always exist, but HostSpace does.

Reply:
done